CMakeLists.txt

```diff
@@ -145,6 +153,8 @@ include (${SHARED_DIR}/utils/uncommon/shared_uncommon.cmake)
 set (THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
 
+add_definitions(-DCOMPILING_WASM_RUNTIME_API=1)
```

Comment:
This macro is only for windows build, is it better to add a control for it, e.g. `if (MSVC) ... endif()`?

Reply:
BTW, could you rebase this PR with wamr main branch? Since a CI error was fixed: https://github.com/bytecodealliance/wasm-micro-runtime/commit/0b62cc89218030561189f65cc42585319e2ed38c

Reply:
fixed & rebased